overlord/snapstate/check_snap.go

```diff
@@ -594,6 +594,45 @@ func checkConfigureHooks(_ *state.State, snapInfo, curInfo *snap.Info, _ snap.Co
 	return nil
 }
 
+func checkDesktopFileIDsConflicts(st *state.State, info *snap.Info) error {
+	desktopFileIDs, err := info.DesktopPlugFileIDs()
+	if err != nil {
+		return err
+	}
+
+	if len(desktopFileIDs) == 0 {
+		return nil
+	}
+
+	stateMap, err := All(st)
+	if err != nil {
+		return err
+	}
+	for instanceName, snapst := range stateMap {
+		if instanceName == info.InstanceName() {
+			continue
+		}
+
+		otherInfo, err := snapst.CurrentInfo()
+		if err != nil {
+			return err
+		}
+
+		otherDesktopFileIDs, err := otherInfo.DesktopPlugFileIDs()
+		if err != nil {
+			return err
+		}
+		for _, desktopFileID := range desktopFileIDs {
+			for _, otherDesktopFileID := range otherDesktopFileIDs {
+				if desktopFileID == otherDesktopFileID {
+					return fmt.Errorf("snap %q requesting desktop-file-id %q conflicts with snap %q use", info.InstanceName(), desktopFileID, instanceName)
+				}
+			}
+		}
```

Comment:
This could use `str.ListContains` instead to be a bit simpler.
Maybe it's not worth it because the number of ids is likely short but we could also put `desktopFileIDs` in a map before looping and use that to do this check so this wouldn't be 3 nested loops. But again, maybe it's not worth the effort given the likely sizes

Reply:
I thought it would be faster to just loop through given it is going to be just a few desktop ids at best because it would skip allocation.

Reply:
switched to `str.ListContains`, Thanks!